platform/src/main/java/org/hillview/storage/CassandraSSTableLoader.java

```diff
@@ -0,0 +1,309 @@
+import java.util.stream.Stream;
+import java.io.IOException;
+import java.util.*;
+import java.util.stream.StreamSupport;
+
+import org.apache.cassandra.db.PartitionColumns;
+import org.apache.cassandra.db.SerializationHeader;
+import org.apache.cassandra.db.marshal.AbstractType;
+import org.apache.cassandra.db.marshal.UTF8Type;
+import org.apache.cassandra.dht.*;
+import org.apache.cassandra.io.sstable.metadata.MetadataComponent;
+import org.apache.cassandra.io.sstable.metadata.MetadataType;
+import org.apache.cassandra.utils.FBUtilities;
+
+/**
+ * Knows how to read a local Cassandra's SSTable file.
+ */
+public class CassandraSSTableLoader {
+    @Nullable
+    private Schema actualSchema;
+    @Nullable
+    private final String ssTablePath;
+    @Nullable
+    IAppendableColumn[] columns;
+    @Nullable
+    private String currentToken;
+
+    private int currentColumn;
+    private int currentRow;
+    private boolean allowFewerColumns;
+
+    static
+    {
+        /** Initializing DatabaseDescriptor that contains Cassandra's general configuration */
+        DatabaseDescriptor.clientInitialization(false);
+    }
+    
+    public CassandraSSTableLoader(String ssTablePath) {
+        this.ssTablePath = ssTablePath;
+    }
+
+    public CFMetaData getSSTableMetadata(Descriptor desc) throws IOException
+    {
+        if (!desc.version.storeRows())
+            throw new IOException("pre-3.0 SSTable is not supported.");
+
+        EnumSet<MetadataType> types = EnumSet.of(MetadataType.STATS, MetadataType.HEADER);
+        Map<MetadataType, MetadataComponent> sstableMetadata = desc.getMetadataSerializer().deserialize(desc, types);
+        SerializationHeader.Component header = (SerializationHeader.Component) sstableMetadata.get(MetadataType.HEADER);
+        IPartitioner partitioner = FBUtilities.newPartitioner(desc);
+
+        CFMetaData.Builder builder = CFMetaData.Builder.create("keyspace", "table").withPartitioner(partitioner);
+        header.getStaticColumns().entrySet().stream()
+                .forEach(entry -> {
+                    ColumnIdentifier ident = ColumnIdentifier.getInterned(UTF8Type.instance.getString(entry.getKey()), true);
+                    builder.addStaticColumn(ident, entry.getValue());
+                });
+        header.getRegularColumns().entrySet().stream()
+                .forEach(entry -> {
+                    ColumnIdentifier ident = ColumnIdentifier.getInterned(UTF8Type.instance.getString(entry.getKey()), true);
+                    builder.addRegularColumn(ident, entry.getValue());
+                });
+        builder.addPartitionKey("PartitionKey", header.getKeyType());
+        for (int i = 0; i < header.getClusteringTypes().size(); i++)
+        {
+            builder.addClusteringColumn("clustering" + (i > 0 ? i : ""), header.getClusteringTypes().get(i));
+        }
+        return builder.build();
+    }
+
+    private <T> Stream<T> iterToStream(Iterator<T> iter)
```

Comment:
you can probably eliminate this function.

Reply:
Removed